packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -260,6 +258,16 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 					source: JSON.stringify(config),
 				});
 			},
+			writeBundle() {
+				if (
+					this.environment.name ===
+					(resolvedPluginConfig.type === "assets-only"
+						? "client"
+						: resolvedPluginConfig.entryWorkerEnvironmentName)
+				) {
+					writeDeployConfig(resolvedPluginConfig, resolvedViteConfig);
+				}
```

Comment:
this whole check is here just so that we only built the config once right?
I'm afraid that this can become unclear down the line, I would suggest to make this code a bit more explanatory and maybe add a comment to clarify what's going on, something along these lines:
```suggestion
				const buildEnvironmentTarget =
					resolvedPluginConfig.type === "assets-only"
						? "client"
						: resolvedPluginConfig.entryWorkerEnvironmentName;
				// the deploy config is unique per application (not per environment)
				// this check makes sure that we only run it once
				if (this.environment.name === buildEnvironmentTarget) {
					writeDeployConfig(resolvedPluginConfig, resolvedViteConfig);
				}
```

Reply:
[4212d55](https://github.com/cloudflare/workers-sdk/pull/8079/commits/4212d5523aa8df3d7d2ceb57e3ce4449cfc11799)